File: android/src/main/java/com/plaid/PLKEmbeddedViewManager.java
package com.plaid;

import android.content.Context;
import android.view.LayoutInflater;

import com.facebook.react.common.MapBuilder;
import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;

import java.util.Map;

public class PLKEmbeddedViewManager extends SimpleViewManager<PLKEmbeddedView> {

    public static final String REACT_CLASS = "PLKEmbeddedView";
    public static final String EVENT_NAME = "onEmbeddedEvent";

    @Override
    public String getName() { return REACT_CLASS; }

    @Override
    public PLKEmbeddedView createViewInstance(ThemedReactContext context) {
        PLKEmbeddedView view = new PLKEmbeddedView(context);
        return  view;
    }

    @ReactProp(name ="token")
    public void setToken(PLKEmbeddedView view, String token) {
        view.setToken(token);
    }

    public Map getExportedCustomBubblingEventTypeConstants() {
        String propName = "onEmbeddedEvent";
        Map handler = MapBuilder.of("phasedRegistrationNames",MapBuilder.of("bubbled", propName));

        MapBuilder.Builder events = MapBuilder.builder();
        events.put(PLKEmbeddedViewManager.EVENT_NAME, handler);
        return events.build();
    }
}

File: android/src/main/java/com/plaid/PLKEmbeddedViewManager.kt
package com.plaid

import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp

class PLKEmbeddedViewManager : SimpleViewManager<PLKEmbeddedView>() {
  override fun getName(): String {
    return REACT_CLASS
  }

  public override fun createViewInstance(context: ThemedReactContext): PLKEmbeddedView {
    return PLKEmbeddedView(context)
  }

  @ReactProp(name = "token")
  fun setToken(view: PLKEmbeddedView, token: String?) {
    view.setToken(token ?: "")
  }

  override fun getExportedCustomBubblingEventTypeConstants(): Map<String, Any> {
    return mutableMapOf(
      EVENT_NAME to mutableMapOf(
        "phasedRegistrationNames" to mutableMapOf(
          "bubbled" to EVENT_NAME
        )
    ))
  }

  companion object {
    const val REACT_CLASS = "PLKEmbeddedView"
    const val EVENT_NAME = "onEmbeddedEvent"
  }
}
